include/aspect/material_model/reaction_model/katz2003_mantle_melting.h

```diff
@@ -89,11 +89,47 @@ namespace aspect
                           const double pressure,
                           const double maximum_melt_fraction,
                           const NonlinearDependence::Dependence dependence) const;
+
+
+          /**
+           * Compute all the reaction rate variables needed for a reactive transport model based on the
+           * Katz 2003 formulation. Takes @p model inputs and @p model outputs to return.
+          */
+          void calculate_reaction_rate_outputs(const typename Interface<dim>::MaterialModelInputs &in,
+                                               typename Interface<dim>::MaterialModelOutputs &out) const;
+
+          /**
+           * Compute all the fluid variables needed for a reactive transport model based on the
+           * Katz 2003 formulation. Takes @p model inputs and @p model outputs to return.
+          */
+          void calculate_fluid_outputs(const typename Interface<dim>::MaterialModelInputs &in,
+                                       typename Interface<dim>::MaterialModelOutputs &out,
+                                       const double thermal_expansivity,
+                                       const double reference_T) const;
+
+          double get_reference_permeability() const;
+
+          double get_eta_f() const;
+
         private:
           /**
           * Parameters for anhydrous melting of peridotite after Katz, 2003
           */
 
+          double reference_rho_f;
```

Comment:
```suggestion
          double reference_rho_fluid;
```

Reply:
I talked to @jdannberg about `out.entropy_derivative_pressure`, because that's part of the reaction rate calculation, and for that calculation I also need `out.reaction_terms` we decided to just keep passing `out` into the function.